input/docs/handbook/data-persistence/index.md

```diff
@@ -29,58 +36,100 @@ public class SearchViewModel : ReactiveObject, ISearchViewModel
+Create a class that implements the `ISuspensionDriver` interface. There are several production-ready implementations below, especially the Akavache suspension driver that works on any platform supported by ReactiveUI. We highly recommend using [Akavache](https://github.com/reactiveui/akavache) suspension driver. Xamarin.iOS and Universal Windows Platform will replicate [Akavache](https://github.com/reactiveui/akavache) data to the cloud and synchronize it to all user devices on which the app is installed!
 
-### SuspensionDriver
+<details><summary>Akavache Platform Independent Suspension Driver</summary>
+<p>
 
-Here is an implementation that uses [Akavache](https://github.com/reactiveui/Akavache) for its persistense. The SuspensionDriver is platform independent, tested on iOS, Android, WPF, UWP, etc. 
+Here is an implementation that uses [Akavache](https://github.com/reactiveui/Akavache) for its persistense. The `SuspensionDriver` is platform independent, tested on iOS, Android, WPF, UWP, etc. 
 
 ```cs
 public class AkavacheSuspensionDriver<TAppState> : ISuspensionDriver where TAppState : class
 {
-    const string appStateKey = "appState";
+    private const string AppStateKey = "appState";
   
     public AkavacheSuspensionDriver() => BlobCache.ApplicationName = "Your Application Name";
 
-    public IObservable<Unit> InvalidateState() => BlobCache.UserAccount.InvalidateObject<TAppState>(appStateKey);
+    public IObservable<Unit> InvalidateState() => BlobCache.UserAccount.InvalidateObject<TAppState>(AppStateKey);
   
-    public IObservable<object> LoadState() => BlobCache.UserAccount.GetObject<TAppState>(appStateKey);
+    public IObservable<object> LoadState() => BlobCache.UserAccount.GetObject<TAppState>(AppStateKey);
 
-    public IObservable<Unit> SaveState(object state) => BlobCache.UserAccount.InsertObject(appStateKey, (TAppState)state);
+    public IObservable<Unit> SaveState(object state) => BlobCache.UserAccount.InsertObject(AppStateKey, (TAppState)state);
 }
 ```
 
-### AppState
+</p>
+</details>
 
-The AppState is an object with `DataContract` or `Newtonsoft.Json` notations, the AppState is platform independent.
+<details><summary>Newtonsoft.JSON Suspension Driver</summary>
+<p>
+
+Here is an implementation that uses [Newtonsoft.Json](https://github.com/JamesNK/Newtonsoft.Json) with `TypeNameHandling.All` json serialization setting for its persistense. The type information is included into the serialized JSON file, which means that `RoutingState` navigation stack consisting of `IRoutableViewModel`s could be restored. However, this driver isn't compatible with UWP. If you'd like to persist UWP state to a file, use `StorageFile` APIs instead of `System.IO.File`.
```

Comment:
Wonder if we can mention maybe the xamarin essentials secure storage here. I haven't really tested in this scenario though so don't stop the pr for this.

Reply:
Yeah, probably worth adding more suspension drivers e.g. `xml` serializer which can store specific type information (required for deserializing `IRoutableViewModel` navigation stack), add more info on how to implement a driver which stores application state on a web server using `Refit` etc.